src/utils/calendar/recurrenceUtil.ts

```diff
@@ -0,0 +1,77 @@
+export type APIRecurrenceRuleFrequency = 0 | 1 | 2 | 3;
+export type APIRecurrenceRuleWeekDay = 0 | 1 | 2 | 3 | 4 | 5 | 6;
+export type APIRecurrenceRuleMonth =
+  | 1
+  | 2
+  | 3
+  | 4
+  | 5
+  | 6
+  | 7
+  | 8
+  | 9
+  | 10
+  | 11
+  | 12;
+
+export interface APIRecurrenceRule {
+  start: Date;
+  end?: Date | undefined | null;
+  frequency: APIRecurrenceRuleFrequency;
+  interval: number;
+  by_weekday?: APIRecurrenceRuleWeekDay[] | undefined | null;
+  by_n_weekday?:
+    | { n: 1 | 2 | 3 | 4 | 5; day: APIRecurrenceRuleWeekDay }[]
+    | undefined
+    | null;
+  by_month?: APIRecurrenceRuleMonth[] | undefined | null;
+  by_month_day?: number[] | undefined | null;
+  by_year_day?: number[] | undefined | null;
+  count?: number | undefined | null;
```

Comment:
フィールドのお名前に `?` 付けたら `undefined` はなくても自動的につくよ
`null` も要件がないなら（シリアライズしたいとか？）でなければ `undefined` でいいかなーと思ったり
```suggestion
  end?: Date | null;
  frequency: APIRecurrenceRuleFrequency;
  interval: number;
  by_weekday?: APIRecurrenceRuleWeekDay[] | null;
  by_n_weekday?:
    | { n: 1 | 2 | 3 | 4 | 5; day: APIRecurrenceRuleWeekDay }[]
    | null;
  by_month?: APIRecurrenceRuleMonth[] | null;
  by_month_day?: number[] | null;
  by_year_day?: number[] | null;
  count?: number | null;
```